File: studio/src/main/captions.ts
// Parses the reel/post caption sheets: sections headed "=== ... — TICKER (..) ===",
// returns { TICKER: "caption + hashtags block" }.
export function parseCaptions(text: string): Record<string, string> {
  const out: Record<string, string> = {}
  if (!text.trim()) return out
  const blocks = text.split(/^={3,}.*$/m)
  const heads = text.match(/^={3,}.*$/gm) || []
  for (let i = 0; i < heads.length; i++) {
    const tk = heads[i].match(/—\s*([A-Z]{1,6})\b/)?.[1]
    if (!tk) continue
    const body = (blocks[i + 1] || '').trim()
    if (body) out[tk] = body
  }
  return out
}

File: review/captions.py
"""Parse reel/post caption sheets: sections headed '=== ... — TICKER (..) ===',
returns { TICKER: 'caption + hashtags block' }."""
from __future__ import annotations

import re


def parse_captions(text: str) -> dict[str, str]:
    out: dict[str, str] = {}
    if not text.strip():
        return out
    blocks = re.split(r"(?m)^={3,}.*$", text)
    heads = re.findall(r"(?m)^={3,}.*$", text)
    for i, head in enumerate(heads):
        m = re.search(r"—\s*([A-Z]{1,16})\b", head)
        if not m:
            continue
        body = (blocks[i + 1] if i + 1 < len(blocks) else "").strip()
        if body:
            out[m.group(1)] = body
    return out
